dynawo/sources/Modeler/Common/DYNDynamicData.cpp

```diff
@@ -309,38 +280,57 @@ DynamicData::getParametersSet(const string& modelId, const string& parFile, cons
 
 void
 DynamicData::createModelDescriptions() {
+  modelicaModels_.clear();
+  blackBoxModels_.clear();
+  modelTemplateExpansions_.clear();
+  modelTemplates_.clear();
+
   for (dynamicdata::dynamicModel_iterator itModel = dynamicModelsCollection_->beginModel();
           itModel != dynamicModelsCollection_->endModel();
           ++itModel) {
     shared_ptr<ModelDescription> model = shared_ptr<ModelDescription>(new ModelDescription(*itModel));
     string staticId = "";
+
+    // init value will be replaced after
+    std::map<std::string, boost::shared_ptr<DYN::ModelDescription> >* mappedModel = NULL;
+
     switch ((*itModel)->getType()) {
       case Model::MODELICA_MODEL: {
         shared_ptr<ModelicaModel> modelicaModel = dynamic_pointer_cast<ModelicaModel>(*itModel);
         staticId = modelicaModel->getStaticId();
+        mappedModel = &modelicaModels_;
         break;
       }
       case Model::MODEL_TEMPLATE:
+        mappedModel = &modelTemplates_;
         break;
       case Model::BLACK_BOX_MODEL: {
         shared_ptr<BlackBoxModel> bbm = dynamic_pointer_cast<BlackBoxModel>(*itModel);
         staticId = bbm->getStaticId();
+        mappedModel = &blackBoxModels_;
         break;
       }
       case Model::MODEL_TEMPLATE_EXPANSION: {
         shared_ptr<ModelTemplateExpansion> modelTemplateExp = dynamic_pointer_cast<ModelTemplateExpansion>(*itModel);
         staticId = modelTemplateExp->getStaticId();
+        mappedModel = &modelTemplateExpansions_;
         break;
       }
     }
     model->setStaticId(staticId);
     if (staticId != "" && dataInterface_) {
       dataInterface_->hasDynamicModel(staticId);
     }
-    modelDescriptions_[model->getID()] = model;
+    std::pair<std::string, boost::shared_ptr<DYN::ModelDescription> > item = std::make_pair(model->getID(), model);
+    modelDescriptions_.insert(item);
+
+    if (mappedModel != NULL) {
+      mappedModel->insert(item);
+    }
   }
 
-  // @todo: instead of copying the full table one by one, directly copy the vector ?
+
+  // @todo: instead of copying the full table one by one, directly copy the vector ? => cannot be done by design of the iterators
```

Comment:
IHMO, I wouldn't keep the todo but will only keep a comment like this "One by one copy due to the iterators design (no other possibility)"